src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -166,7 +164,15 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   _onTouched = () => {};
 
   /** The autocomplete panel to be attached to this trigger. */
-  @Input('matAutocomplete') autocomplete: MatAutocomplete;
+  @Input('matAutocomplete')
+  get autocomplete(): MatAutocomplete { return this._autocomplete; }
+  set autocomplete(value: MatAutocomplete) {
+    this._autocomplete = value;
+    if (this._overlayRef) {
+      this._overlayRef.detach();
```

Comment:
@crisbeto  Not sure if it would be better to do call `closePanel()` here instead of just `detach` so the values of `_overlayAttached` and `autocomplete._isOpen` are getting updated as well.